sdks/python/apache_beam/transforms/core.py

```diff
@@ -988,10 +991,14 @@ def apply(self, elements, *args, **kwargs):
       *args: Additional arguments and side inputs.
       **kwargs: Additional arguments and side inputs.
     """
+    if args or kwargs:
+      add_input = lambda accumulator, element: self.add_input(
+          accumulator, element, *args, **kwargs)
+    else:
```

Comment:
am I missing something or the else clause is not necessary?

Reply:
It's for performance reasons. Most of the time args and kwargs are empty (and it is much cheap to not have the level of indirection or use them).